### PR TITLE
Improve InterAppCalls documentation + update to current version

### PR DESCRIPTION
**Pull request by MichielOda**

### CODE REVIEW

File: develop/devguide/Core.InterAppCalls/InterAppCalls_GettingStarted_CreatingExecutor.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+In other cases, you want to immediately create a message that will eventually need to be returned to the sender.
+
+```csharp
+public override Message CreateReturnMessage()
+{
+    return new MyResponse { Guid = Message.Guid, Success = true };
+}
+```
+
```
Comment (janstaelensskyline) [inline develop/devguide/Core.InterAppCalls/InterAppCalls_GettingStarted_CreatingExecutor.md:98]: The Executor is following a design pattern called **Template Method Pattern**

Benefits:
Consistency: The overall process is consistent and follows the defined sequence.
Flexibility: Specific steps can be overridden in subclasses if different behavior is needed for certain executors.
Clarity: The sequence of steps is clearly defined in one place (the execute method of Executor).
This design pattern ensures that the core structure of the execution process remains intact while allowing for customization and extension where necessary.

If your logic does not require the standard Method Template, consider using the [simple executor](xref:InterAppCalls_Customizations#creating-a-simple-executor). This approach is ideal for simpler situations and helps avoid code bloat.

File: develop/devguide/Core.InterAppCalls/InterAppCalls_GettingStarted_CreatingExecutor.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+> [!NOTE]
+> A return message does not necessarily need to be something to send to an external destination. A message could also be part of an internal API used to move data between classes, methods or QActions within your own connector. This can also be returned.
+
+> [!TIP]
+> See the [simple executor](xref:InterAppCalls_Customizations#creating-a-simple-executor) if more simplicity is needed.
```
Comment (janstaelensskyline) [inline develop/devguide/Core.InterAppCalls/InterAppCalls_GettingStarted_CreatingExecutor.md:103]:
```suggestion
>If your logic does not require the standard Method Template, consider using the  [simple executor](xref:InterAppCalls_Customizations#creating-a-simple-executor). This approach is ideal for simpler situations and helps avoid code bloat.
```

**Review by janstaelensskyline — CHANGES_REQUESTED** [review]

File: develop/devguide/Core.InterAppCalls/InterAppCalls_GettingStarted_CreatingExecutor.md (hunk 1 above)
Comment (MichielOda) [reply develop/devguide/Core.InterAppCalls/InterAppCalls_GettingStarted_CreatingExecutor.md:98]: Done

File: develop/devguide/Core.InterAppCalls/InterAppCalls_GettingStarted_ReceivingCall.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
+---
+uid: InterAppCalls_GettingStarted_ReceivingCall
+---
+
+# Receiving a call
+
+You can create a QAction that triggers on parameter `interApp_receive`. This QAction will handle any incoming call. The code here will be easy as it uses a combination of the factory and command design patterns.
+
+You get the raw value from the parameter and you use the `InterAppCallFactory` to turn it into an `IInterAppCall`. In addition the known types need to be added so that the factory can dedeuce which class it needs to deserialize to. More info about known types can be found here: [Known Types](xref:InterAppCalls_KnownTypes).
```
[the remark this answers is not in the document]
Comment (MichielOda) [reply develop/devguide/Core.InterAppCalls/InterAppCalls_GettingStarted_ReceivingCall.md:9]: Done

File: develop/devguide/Core.InterAppCalls/InterAppCalls_GettingStarted_ReceivingCall.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+> [!IMPORTANT]
+> It is not possible to receive both single messages and InterApp calls on the same parameter. We recommend just sticking to the InterApp call.
+
+## Receiving a call without replying
+
+```csharp
+    public static void HandleIncomingMessages(SLProtocol protocol)
+    {
+        // Prepare necessary mappings
+        List<Type> knownTypes = new List<Type> { typeof(MyMessage), typeof(MyResponse) };
```
[the remark this answers is not in the document]
Comment (MichielOda) [reply develop/devguide/Core.InterAppCalls/InterAppCalls_GettingStarted_ReceivingCall.md:20]: Done

File: develop/devguide/Core.InterAppCalls/InterAppCalls_GettingStarted_SendingCall.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+# Sending a call
+
+## Sending a call without expected return
+
+```csharp
+    public static void SendMessages(SLProtocol protocol, int dmaId, int elementId)
+    {
+        // Prepare necessary mappings
+        List<Type> knownTypes = new List<Type> { typeof(MyMessage), typeof(MyResponse) };
```
[the remark this answers is not in the document]
Comment (MichielOda) [reply develop/devguide/Core.InterAppCalls/InterAppCalls_GettingStarted_SendingCall.md:13]: Done

File: develop/devguide/Core.InterAppCalls/InterAppCalls_Introduction.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+- A user-friendly and flexible way to define any class you want as a message.
+- The ability to balance maintainability versus messaging speed as needed for your project.
+
+Its main purpose is for use within large projects where inter-element or inter-automation communication is required, where you can adjust source and destination code to create and parse the messages.
+
+## Requirements
+
+### DataMiner requirements
+
+- The InterApp classes require DataMiner 10.1.0 or higher to function correctly.
```
[the remark this answers is not in the document]
Comment (MichielOda) [reply develop/devguide/Core.InterAppCalls/InterAppCalls_Introduction.md:28]: Done

File: develop/devguide/Core.InterAppCalls/InterAppCalls_Introduction.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+### Automation script requirements
+
+- DataMiner version 10.1.0 or higher is required.
```
[the remark this answers is not in the document]
Comment (MichielOda) [reply develop/devguide/Core.InterAppCalls/InterAppCalls_Introduction.md:32]: Done

File: develop/devguide/Core.InterAppCalls/InterAppCalls_Introduction.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+### Protocol requirements
```
[the remark this answers is not in the document]
Comment (MichielOda) [reply develop/devguide/Core.InterAppCalls/InterAppCalls_Introduction.md:36]: Done

File: develop/devguide/Core.InterAppCalls/InterAppCalls_GettingStarted.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
+---
+uid: InterAppCalls_GettingStarted
+---
+
+# Getting Started
+
```
[the remark this answers is not in the document]
Comment (MichielOda) [reply develop/devguide/Core.InterAppCalls/InterAppCalls_GettingStarted.md:6]: @SimonV-Skyline, @ArneMaes0: Anything you want to add here? Or would you prefer to have this is another PR?

[the remark this answers is not in the document]
Comment (MichielOda) [reply develop/devguide/Core.InterAppCalls/InterAppCalls_GettingStarted.md:6]: As discussed in chat, I've updated the examples with the 3 repo's that I found. As the examples on GitHub are still WIP, these pages can be updated later when needed.

File: develop/devguide/Core.InterAppCalls/InterAppCalls_GettingStarted.md [hunk 10]
```diff
@@ -5,7 +5,7 @@ uid: InterAppCalls_GettingStarted
 # Getting Started
 
 > [!NOTE]
-> An example connector "Skyline Example InterAppCalls" is available in the [SkylineCommunications/SLC-C-Example_InterAppCalls](https://github.com/SkylineCommunications/SLC-C-Example_InterAppCalls) GitHub repository.
+> An example is available in the [SkylineCommunications/SLC-S-Example_InterAppCalls](https://github.com/SkylineCommunications/SLC-S-Example_InterAppCalls) GitHub repository.
```
Comment (janstaelensskyline) [inline develop/devguide/Core.InterAppCalls/InterAppCalls_GettingStarted.md:8]: This is a solution and not a connector. We probably want to still point to the connector that shows how to use the interapp.

Comment (MichielOda) [reply develop/devguide/Core.InterAppCalls/InterAppCalls_GettingStarted.md]: I've updated it to point to all three.

**Comment by MichielOda** [comment]
As discussed with @janstaelensskyline, I'll leave out the references to the GitHub repositories as not everything was properly migrated from Jenkins to GitHub and @ArneMaes0 & @SimonV-Skyline are busy modifying the repositories with new code.
When they are finished, then can add the references again with the latest changes that they make.

**Review by janstaelensskyline — APPROVED** [review]

**Review by MariekeGO — COMMENTED** [review]
I've already made several changes during my initial review, but some comments remain that  should be looked into. Could you take a look, @MichielOda?
Some notes about the changes in my initial review:
- I've changed "Introduction" to "About the InterApp framework", as this is a more meaningful title for someone who lands on the page from the search.
- I've removed the link to the release notes on that introduction page, as we don't usually add such links in the documentation, and adding them will cause unnecessary extra update work (which is likely to get forgotten) whenever new release notes are published. 
- I've also removed the links to the RN about "Enhanced efficiency on InterApp reply calls", because generally linking to release notes for more information is not a good idea in the user guide or dev docs. If the feature evolves and the initial release note is no longer applicable, that link will lead to outdated info because the release note itself will never be updated. It's better to immediately fully document the relevant information where appropriate.
- I've moved the connector requirements so all requirements are listed under the same "Requirements" header. Otherwise, spreading out the requirements over several places could cause some of them to be overlooked, since many users do not read the documentation page by page but instead skip to specific pages.

File: develop/devguide/Core.InterAppCalls/InterAppCalls_GettingStarted.md [hunk 11]
```diff
@@ -0,0 +1,14 @@
+---
+uid: InterAppCalls_GettingStarted
+---
+
+# Getting started with the InterApp framework
+
+Begin by checking and making sure the [DataMiner requirements](xref:InterAppCalls_Introduction#dataminer-requirements), DIS requirements, [connector requirements](xref:InterAppCalls_Introduction#connector-requirements), and optional Automation script requirements are met.
```
[the remark this answers is not in the document]
Comment (MichielOda) [reply develop/devguide/Core.InterAppCalls/InterAppCalls_GettingStarted.md:7]: I've removed those 2 as they aren't really relevant. InterAppCalls don't have any requirement of DIS (unless the NuGet support was meant, but that is irrelevant as we don't support versions without that anymore). For Automation script, it was the same as the dataminer requirements so not needed as well.

File: develop/devguide/Core.InterAppCalls/InterAppCalls_Customizations.md [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+
+## Custom serializer
+
+The *InterAppCall* library uses a special serializer that has been deliberately tweaked, adjusted, and heavily tested to work with almost any possible scenario. However, it also supports developers writing their own serializers and using those.
+
+Any factory method that creates an InterApp call or message and any method to send the InterApp call or message has an overload that takes a class implementing the `Skyline.DataMiner.Core.InterAppCalls.Common.Serializing.ISerializer` interface.
+
+This interface requires one method to serialize and one method to deserialize. You can create your own logic to do this and provide this to your methods, and it will override the internal serializer with your own logic.
+
+## Creating a simple executor
```
[the remark this answers is not in the document]
Comment (MichielOda) [reply develop/devguide/Core.InterAppCalls/InterAppCalls_Customizations.md:15]: At the bottom of the 'Creating Executor' page, you can find a link to the simple executor with the explanation. I've added a similar TIP at the bottom of the simple executor.

**Review by MariekeGO — APPROVED** [review]

**Pull request merged**